tests/reg_tests/test_solver_combos.py

```diff
@@ -0,0 +1,217 @@
+# built-ins
+import unittest
+import os
+import copy
+import numpy
+from parameterized import parameterized_class
+
+# MACH classes
+from adflow import ADFLOW
+from adflow import ADFLOW_C
+
+# import the testing utilities that live a few directories up
+import reg_test_utils as utils
+from reg_default_options import adflowDefOpts
+from reg_aeroproblems import ap_simple_cart_cube
+import reg_test_classes
+
+
+baseDir = os.path.dirname(os.path.abspath(__file__))
+
+# tests for different solver combinations using a simple cartesian block.
+# we are interested in these tests converging without seg-faulting
+# and switching between the solvers correctly. We test
+# Euler + laminar NS + RANS, with smoother, ANK, SANK, CSANK ,NK transition order
+# with turbDADI and turbKSP. So 3 eqn types and 2 turbulence solvers, we have 6 tests
+# for both real and complex cases. These do not cover every possible combination, but
+# represent the most common use cases. Testing turbulence solver options with euler
+# and laminar NS makes sure the solver does not segfault etc.
+
+solver_combo_params = [
+    # Euler Tests
+    {
+        "name": "euler_smoother_ank_sank_csank_nk_turbdadi",
+        "options": {
+            "equationtype": "Euler",
+            # euler fails to get to machine zero in parallel...
```

Comment:
Why is that?

Reply:
I dont know. The linear solver result changed enough to trip the solver. Euler cases are more challenging for us to converge now because they dont have as much dissipation as the RANS ones. With that particular tuning, the Newton just stalled about that convergence level. I can probably just play around and make it converge more but it will probably require changing the linear solver tolerance in Newton manually etc. Do you want me to make this case converge or are you okay with the current state @joanibal ?

Reply:
> Do you want me to make this case converge or are you okay with the current state @joanibal ?

This seems important to verify the derivatives to the highest possible accuracy. 
With that being said, I don't want to hold this up. Maybe it's best to merge and add an issue.

Reply:
I see, that makes sense. I will look into this. Worst case, we can have this case converge with 1 proc and that does converge to machine zero. I currently do not check derivatives with these tests but I am guessing you want to extend these tests to check the derivatives? 

As I previously said, I think checking total derivatives on a case where you know the numbers are reasonable makes more sense to me. So thats why I am fine with the adjoint/complex-step comparisons with the wing cases, and then this block test only looks at complex convergence. It is fine if you want to extend the test to complex step derivatives though.

Reply:
Well, thinking about the 2x2x2 cube again, does it even make sense to run that on 2 procs? We would end up getting two slices of the cube on each proc, but one of the directions would only have one layers. I am not sure if this would cause issues with the underlying double-halo communication and BC stuff. Should we just change this test to a 4x4x4 cube?

Reply:
So I just refined the 2x2x2 cube to get a 4x4x4 cube. All tests converge to machine zero w/o issues this way. So @sseraj's suggestion of doing a larger mesh is also satisfied this way. I will update the input_files archive and remove the custom l2 convergence for the Euler tests. 

I will probably rename the cube so that it is obvious that it is a 4^3 cube. I am guessing you will use the 2^3 cube for your own development @joanibal and I dont want to cause any confusion there. 

Does this make sense? Is this the last issue in this PR regarding the tests?